### ISSUE TITLE
Add tests: Complex use of torch tensors

### ISSUE THREAD

**Issue by jhelferty-nv** [shader-slang/slangpy#733]
Chris' comment from Discord:

> one set of things I'm nervous about is how more complex use of torch tensors works - i.e. passing none-contigous torch tensors, or torch tensors that are slices of others into slangpy. Have some test cases that exercise more 'real world' use of pytorch would be well worth it. Right now our tests verify the pytorch integration works but they certainly don't exercise it in anger
> A test case that operates on a sliced up torch tensor, or another that implements a loss function to stick on the end of some torch linear layers, or a bespoke slang adam optimizer applied to the parameters of a torch NN would all be valuable things that represent our user's predicted workloads.
